src/AzureClient/Extensions.cs

```diff
@@ -89,7 +87,11 @@ public static void AddAzureClient(this IServiceCollection services)
             (job.Details.Name != null && job.Details.Name.Contains(filter, StringComparison.OrdinalIgnoreCase)) ||
             (job.Details.Target != null && job.Details.Target.Contains(filter, StringComparison.OrdinalIgnoreCase));
 
-        internal static void Log(this Exception ex, IChannel? channel, ILogger? logger, string msg = "")
+        /// <summary>
+        ///      Writes an exception to a logger and emits it to the stderr stream
+        ///      of a Jupyter display channel.
+        /// </summary>
+        public static void Log(this Exception ex, IChannel? channel, ILogger? logger, string msg = "")
```

Comment:
I noticed this changed from `internal` to `public`. Is this method now being referenced outside this project? If so, wouldn't it be better to make this implementation part of another project such as `Core`?

Reply:
It is being used here in QirMagic.cs: https://github.com/microsoft/iqsharp/blob/36f2ebffe60bf66261435c4dad99e96d47c833f2/src/Kernel/Magic/QirMagic.cs#L81

Reply:
I'm not sure where in the Core would be a good place to put it.

Reply:
In that case I think instead of making this public, the `Logger` object should be used directly instead of an extension method on the `Exception` class. Another option would be to create a `Common` project where these types of extension methods are implemented.

Reply:
Okay I've removed the external reference to Log and just had it call the Logger object directly.

Reply:
And the access has been set back to `internal` for Log.